app/Http/Controllers/ProductController.php

```diff
@@ -0,0 +1,139 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\Category;
+use App\Models\Product;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\DB;
+
+
+class ProductController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+         $products = DB::table('products')
+        ->get();
+        return view('view-products', compact('products'));
+    }
+
+    /**
+     * Show the form for creating a new resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function create()
+    {
+        $categories = Category::all();
+        return view('upload-product',compact('categories'));
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(Request $request)
+    {
+        $request->validate([
+            'name' => ['required'],
+            'description' => ['required'],
+            'price' => ['required'],
+            'category' => ['required'],
+            'path' => ['required','mimes:jpg,jpeg,gif,png,pdf']
```

Comment:
Remember to add units of measurement. I think you'll have to create a separate table like category, to constrain the options the vendors have for this. Same way you did with categories